### PR TITLE
Add trial.py with basic sketch of Trial class.

### PR DESCRIPTION
**Pull request by MartinHeroux**
Again, wanted to flesh out a few of the pieces that will be needed. 

Each instance of the `Trial` class will represent a real-world experimental trial. It will contain some basic information about the trial. It will also contained the imported channel data (method template created but not yet implemented).

I decided to not go any further because I was not 100% certain how to proceed. 

-----------------------

In my previous version of this code, the various channels were located in a dictionary, which was an attribute of the Trial class. For example, you would access the 'knee_torque' channel as follows:

```python
max_ext1.signals['knee_torque']
```  
[Note: My new version of the code has replaced 'signals' with 'channels' to better reflect the terminology used by Spike2.]

I found my previous interface clunky. Because the channels were in a dict, you could not used code completion to access the individual channels. That is, you could not hit Tab and get suggestions from the IDE; you had to remember the exact name of the channel.

In this new iteration of the code, I would like the channels to become attributes of the class. This would allow:

max_ext1.<Tab>

Which, as is typical, would suggest all Class attributes and methods, including the available channels. For example:

```python
max_ext1.knee_torque
```

When I wrote my first version of this code, I did not know about `settatr`, which if I am correct, will allow me to set the attribute name to the actual channel name (e.g. 'knee_torque').

Does this approach seem sensible/feasible?

### CODE REVIEW

**Comment by bbelderbos** [comment]
Maybe worth looking at `requests` / Reitz' work because I think his libraries often support attribute access which indeed is way more elegant than dict keys. So yes I am onboard with this from a conceptual standpoint (would have to see the code though)

File: spike2py/trial.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        subject_id: str
+            Subject's study identifier
+        channels: list
+            List of channel names, as they appeared in the original .smr file.
+            Example: ['biceps', 'triceps', 'torque']
+            If not included, all channels will be processed.
+
+        """
+        if name is None:
+            name = os.path.split(file)[-1].split('.')[0]
```
Comment (bbelderbos) [inline spike2py/trial.py:25]: why 2x `split`?

File: spike2py/trial.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import os
+
+
+class Trial:
+    """Class for experimental trial recorded using Spike2."""
+
+    def __init__(self, file, name=None, subject_id=None, channels='All'):
```
Comment (bbelderbos) [inline spike2py/trial.py:7]: I think 'All' is not the most idiomatic way to do this, what about refactoring this to `channel=None` in which case all will be processed

**Comment by MartinHeroux** [comment]
haha! I initially had it to `None`, but I thought that reading `channels=None` as the default would have the semi-knowledgable reader of the code interpret it that the default behaviour is to include none of the channels. Hence why I changed it to 'All', which reads better in the function call `channels='All`.

But that is clearly not as pythonic. I will change it back.

**Comment by bbelderbos** [comment]
I feel `None` is better here and mirrors the other args as well. Does it make sense to make `channel` singular (not plural)? Would that mean you have to reverse the logic?

File: spike2py/trial.py (hunk 1 above)
Comment (MartinHeroux) [reply spike2py/trial.py:25]: The first split gives the filename `knee_max.mat`, but within the code I want the name of the trial to be `knee_max`, so I trip off the file extension using a split.

**Comment by MartinHeroux** [comment]
I can see you point.

Most of the time we record the channels we want and need, so the default of including all channels will likely be what people use 90% of the time. 

There will be times when you want a few channels, in which case you would specify them. 

Very rarely would you have a bunch of channels but only want to process one of them. But in this case it is true that channel (singular) would be better. But in most other times, people will want all channels or a group of channels; this is why I initially went with `channels`.

I will think about this as I work more with the code to see what is most logical.

**Pull request merged**